src/core/Akka.Remote.TestKit/MsgEncoder.cs

```diff
@@ -7,126 +7,154 @@
+                    wrapper.Failure = new Proto.Msg.InjectFailure()
                     {
-                        w.SetFailure(
-                            InjectFailure.CreateBuilder()
-                                .SetFailure(TCP.FailType.Exit)
-                                .SetExitValue(terminate.ShutdownOrExit.ToRight().Value));
-                    }
-                    else if (terminate.ShutdownOrExit.IsLeft && !terminate.ShutdownOrExit.ToLeft().Value)
+                        Failure = Proto.Msg.InjectFailure.Types.FailType.Exit,
+                        ExitValue = terminate.ShutdownOrExit.ToRight().Value
+                    };
+                }
+                else if (terminate.ShutdownOrExit.IsLeft && !terminate.ShutdownOrExit.ToLeft().Value)
+                {
+                    wrapper.Failure = new Proto.Msg.InjectFailure()
                     {
-                        w.SetFailure(
-                            InjectFailure.CreateBuilder()
-                                .SetFailure(TCP.FailType.Shutdown));
-                    }
-                    else
+                        Failure = Proto.Msg.InjectFailure.Types.FailType.Shutdown
+                    };
+                }
+                else
+                {
+                    wrapper.Failure = new Proto.Msg.InjectFailure()
                     {
-                        w.SetFailure(
-                            InjectFailure.CreateBuilder()
-                                .SetFailure(TCP.FailType.ShutdownAbrupt));
-                    }
-
-                })
-                .With<GetAddress>(
-                    address => w.SetAddr(AddressRequest.CreateBuilder().SetNode(address.Node.Name)))
-                .With<AddressReply>(
-                    reply =>
-                        w.SetAddr(
-                            AddressRequest.CreateBuilder()
-                                .SetNode(reply.Node.Name)
-                                .SetAddr(Address2Proto(reply.Addr))))
-                .With<Done>(done => w.SetDone(string.Empty))
-                .Default(obj => w.SetDone(string.Empty));
+                        Failure = Proto.Msg.InjectFailure.Types.FailType.ShutdownAbrupt
+                    };
+                }
+            }
+            else if (message is GetAddress)
+            {
+                var getAddress = (GetAddress)message;
+                wrapper.Addr = new Proto.Msg.AddressRequest { Node = getAddress.Node.Name };
+            }
+            else if (message is AddressReply)
+            {
+                var addressReply = (AddressReply)message;
+                wrapper.Addr = new Proto.Msg.AddressRequest
+                {
+                    Node = addressReply.Node.Name,
+                    Addr = AddressMessageBuilder(addressReply.Addr)
+                };
+            }
+            else if (message is Done)
+            {
+                wrapper.Done = " ";
```

Comment:
Protobuf encoder didn't like an empty string, didn't give it a type `{ done: ""}`, but instead just an empty string `{{ }}`.  Needs at least length > 0, but open to suggestions - tried String.Empty (doesn't work) and "done" which results in `{ "done": "done"}`